### PR TITLE
try to get this to work.

### PR DESCRIPTION
**Pull request by asherf**
linter pass (except for an issue w/ pylint run not having access to some json file)

@cognifloyd I hope this helps!

### CODE REVIEW

**Comment by asherf** [comment]
this is just for reference, feel free to close this PR at any time.

File: pants.toml [hunk 1]
```diff
@@ -49,6 +49,7 @@ inits = true
 
 [black]
 version = "black==20.8b1"
+extra_requirements = ["regex>=2021.3.17"]
```
Comment (cognifloyd) [inline pants.toml:52]: Why do we need this?

File: pants.toml [hunk 2]
```diff
@@ -79,3 +80,12 @@ args = [
...
   "--load-plugins=db_models",
 ]
+
+[subprocess-environment]
+env_vars.add = [
+  # Works around bad `-arch arm64` flag embedded in Xcode 12.x Python interpreters on intel
+  # machines. See: https://github.com/giampaolo/psutil/issues/1832
+  # On macos do:  export ARCHFLAGS=-arch x86_64 
+  # see: https://github.com/pantsbuild/pants/pull/11733
+  "ARCHFLAGS",
```
Comment (cognifloyd) [inline pants.toml:90]: When I hit this issue, I switched over to my MacPorts-installed copies of python. Good to know there's a solution :)

This won't affect any other platforms will it?

**Review by cognifloyd — COMMENTED** [review]
Cool. It's nice to have more people experimenting here.

File: pants.toml (hunk 1 above)
Comment (asherf) [reply pants.toml:52]: this is the config we have in our internal repo... it works.... If I really think about it was probably added to override the pants default setting here, which I think is setuptools (or at least used to be) which caused issues in the past.

File: pants.toml (hunk 2 above)
Comment (asherf) [reply pants.toml:90]: this option just tells pants to pass this env variable (if defined) to subprocesses. it doesn't actually define it.
in CI, pants defines it for mac os builds, see https://github.com/pantsbuild/pants/pull/11733

Comment (asherf) [reply pants.toml:90]: and there is a thread discussion in pants slack: https://pantsbuild.slack.com/archives/C0D7TNJHL/p1617914862144100

File: pants.toml (hunk 1 above)
Comment (asherf) [reply pants.toml:52]: I double checked, there is no longer a need for this (at least with recent pants versions)

**Comment by cognifloyd** [comment]
I've got pylint running now. Hooray! Thanks for your help in here. I'll close this.

**Pull request closed without merging**